### PR TITLE
modify and rename part of the eviction / finalizer implementation

### PR DESCRIPTION
**Pull request by Jutho**
@GunnarFarneback and @Krastanov:  I've modified part of the implementation and renamed `eviction_callback` to `finalizer`. Let me know what you think. If you agree, I will merge and bump the version

### CODE REVIEW

**Comment by Krastanov** [comment]
I do not see anything unreasonable in the changes. Thanks for taking care of this! I would love to update my dependencies to this upcoming version.

**Comment by GunnarFarneback** [comment]
I have no attachment to the names or the implementation details. It's the functionality that counts. But I'll review this to see if I can spot any problems.

File: src/LRUCache.jl [hunk 1]
```diff
@@ -183,57 +181,54 @@ function Base.resize!(lru::LRU{K, V}; maxsize::Integer = lru.maxsize) where {K,
...
-    _notify_evictions!(lru.eviction_callback, evictions)
+    _finalize_evicitions!(lru.finalizer, evictions)
     return lru
 end
 
-function _notify_evictions!(callback, evictions)
-    while !isempty(evictions)
-        key, value = pop!(evictions)
-        callback(key, value)
+function _finalize_evicitions!(finalizer, evictions)
```
Comment (GunnarFarneback) [inline src/LRUCache.jl]: Evictions is misspelled in this function name.

Also the definition of `_no_callback` can be removed, but github won't let me comment on lines outside the immediate context of the diff.

**Comment by GunnarFarneback** [comment]
I assume these changes weren't enough to avoid allocations where no finalizer is used, other than in `delete!` and `pop!`.

File: src/LRUCache.jl (hunk 1 above)
Comment (Jutho) [reply src/LRUCache.jl]: Thanks for carefully checking this. That's what you get from doing "find and replace all" to quickly :-).

**Comment by Jutho** [comment]
Indeed, I also tried to get rid of other allocations, but in the end the only way I could see how to do it was by making the type of `finalizer` an explicit type parameter of `LRU`, and I assumed that would be a breaking change. So I did not want to do that.

I could see one other way, which is to have an empty `Vector{Tuple{K,V}}()` associated with every instance of `LRU`, i.e. have it as an extra field. If `finalizer === nothing`, one can just use this one (and still check for `finalizer === nothing` later on so as not to push anything to it, cause that would again be not thread safe). It did not seem very elegant, but it might be worth considering.

**Comment by Jutho** [comment]
I addressed the comments, but did not go with my proposal of adding an empty `Vector{Tuple{K,V}}()` to the `LRU` structure. I think that in the typical use cases of an LRU cache, the extra cost of allocating an empty vector are negligible compared to the other operations, even for a single insertion etc.

**Comment by Jutho** [comment]
If CI turns green and @GunnarFarneback gives green light, I will merge and register the new version.

**Comment by Jutho** [comment]
v1.3.0 is scheduled to be merged and should land soon!

**Pull request merged**